lib/azure/signature.rb

```diff
@@ -237,10 +236,13 @@ def generate_string(verb, headers, auth_type)
     #
     def canonicalize_resource(uri)
       resource = '/' + account_name + (uri.path.empty? ? '/' : uri.path)
-      params = CGI.parse(uri.query.to_s).map { |k,v| [k.downcase, v] }
-      params.sort_by! { |k,v| k }
-      params.map! { |k,v| '%s:%s' % [k, v.map(&:strip).sort.join(',')] }
-      [resource, *params].join("\n")
+
+      nested = (uri.query_values(Array) || []).sort
```

Comment:
can you give an example of `uri` or the resulting `nested`? Hard to understand what  `string` looks like.